scripts/linter/index.js

```diff
@@ -0,0 +1,92 @@
+const frontmatter = require("remark-frontmatter");
+const parse = require("remark-parse");
+const path = require("path");
+const report = require("vfile-reporter");
+const stringify = require("remark-stringify");
+const unified = require("unified");
+const vfile = require("to-vfile");
+
+const { collectRecipes } = require("./recipes");
+const collectDocs = require("./collect-docs");
+const deprecatedSections = require("./deprecated-sections");
+const missingSections = require("./missing-sections");
+const validRecipe = require("./valid-recipes");
+const yamlLoader = require("./yaml-loader");
+
+/**
+ * Lint a Markdown file from a path and return a `VFile` with attached messages.
+ */
+async function lintMarkdownDoc(mdParser, filePath) {
+    const file = await vfile.read(filePath);
+    await mdParser().process(await file);
+    return file;
+}
+
+/**
+ * Lint a `meta.yaml` file path and return an array of `VFile`s with attached messages.
+ */
+async function lintMetaYamlDoc(mdParser, filePath) {
+    const file = await vfile.read(filePath);
+    const reportedFiles = [];
+
+    file.message(
+        "meta.yaml is deprecated",
+        file,
+        "stumptown-linter:deprecated-meta-yaml"
+    );
+    reportedFiles.push(file);
+
+    const proseFile = await vfile.read(
+        path.join(path.dirname(file.path), "prose.md")
+    );
+
+    // Copy `meta.yaml` contents as frontmatter for `prose.md`.
+    proseFile.contents = [
+        "---",
+        file.contents.toString(),
+        "---",
+        proseFile.contents.toString()
+    ].join("\n");
+
+    await mdParser().process(proseFile);
+    reportedFiles.push(proseFile);
+
+    return reportedFiles;
+}
+
+async function main() {
+    const recipes = await collectRecipes();
+
+    const markdownParser = unified()
+        .use(parse)
+        .use(stringify)
+        .use(frontmatter, ["yaml"])
+        .use(yamlLoader)
+        .use(validRecipe, { recipes })
+        .use(deprecatedSections, {
+            sections: {
+                "html-element": ["usage_notes"],
+                "css-property": ["usage_notes"]
+            }
+        })
+        .use(missingSections);
+
+    const filePaths = await collectDocs();
+
+    const reportedFiles = [];
+
+    for (const fp of filePaths) {
+        if (fp.endsWith("meta.yaml")) {
```

Comment:
I don't think we are expecting to have meta.yaml anywhere in the tree from now on. I see in this code that you are actually treating it as an error, but given that finding this error requires a bunch of special-casing in the linter, I think it would be better just to fix the three CSS files where it occurs:

```
content/css/reference/properties/border/meta.yaml
content/css/reference/properties/border-block-start-width/meta.yaml
content/css/reference/properties/transform/meta.yaml
```

...and remove the checking code from the linter.